### PR TITLE
Specify tags for L1 and L2 interactions

### PR DESCRIPTION
**Pull request by janekolszak**
- Relayer: Fix data item tags
- Relayer: Configurable environment

### CODE REVIEW

File: src/relay/arweave_meta_bundler.go [hunk 1]
```diff
@@ -120,13 +111,21 @@ func (self *ArweaveMetaBundler) createMetaDataItem(payload *Payload, arweaveBloc
...
+		{Name: "Sequencer", Value: "Warp"},
+		{Name: "App-Name", Value: "WarpMeta"},
+		{Name: "Env", Value: self.Config.Relayer.Environment},
+		// Block specific
+		{Name: "Arweave-Block-Height", Value: strconv.FormatUint(arweaveBlock.Message.BlockInfo.Height, 10)},
+		{Name: "Arweave-Block-Timestamp", Value: strconv.FormatUint(arweaveBlock.Message.BlockInfo.Timestamp, 10)},
+		{Name: "Arweave-Block-Hash", Value: arweaveBlock.Message.BlockInfo.Hash},
+		{Name: "Sequencer-Height", Value: strconv.FormatInt(payload.SequencerBlockHeight, 10)},
+		{Name: "Sequencer-Timestamp", Value: strconv.FormatInt(payload.SequencerBlockTimestamp, 10)},
+		// {Name: "Sequencer-Hash", Value: base64.RawURLEncoding.EncodeToString(payload.SequencerBlockHash.Bytes())},
```
Comment (szynwelski) [inline src/relay/arweave_meta_bundler.go:128]: to remove

File: src/relay/arweave_meta_bundler.go [hunk 2]
```diff
@@ -120,13 +111,21 @@ func (self *ArweaveMetaBundler) createMetaDataItem(payload *Payload, arweaveBloc
...
 		{Name: "Bundle-Version", Value: "2.0.0"},
-		{Name: "App-Name", Value: "Warp"},
-		// FIXME: Is this Action OK?
-		{Name: "Action", Value: "WarpMetaData"},
-		// Tags common with L2
-		{Name: "Sequencer", Value: "RedStone"},
+		// Global
+		{Name: "Source", Value: "Warp"},
+		{Name: "Sequencer", Value: "Warp"},
+		{Name: "App-Name", Value: "WarpMeta"},
```
Comment (szynwelski) [inline src/relay/arweave_meta_bundler.go:120]: I wouldn't add it for now

File: src/relay/arweave_meta_bundler.go [hunk 3]
```diff
@@ -120,13 +111,21 @@ func (self *ArweaveMetaBundler) createMetaDataItem(payload *Payload, arweaveBloc
 	}
 
 	out.Tags = bundlr.Tags{
```
Comment (szynwelski) [inline src/relay/arweave_meta_bundler.go:113]: add Contract and Tx-Id

File: src/relay/arweave_meta_bundler.go [hunk 4]
```diff
@@ -120,13 +111,21 @@ func (self *ArweaveMetaBundler) createMetaDataItem(payload *Payload, arweaveBloc
...
+		// https://github.com/Irys-xyz/js-sdk/blob/cdf73fa6bf537c57e6c9050ff0cd7d18ebc2f0ac/src/common/upload.ts#L237
 		{Name: "Bundle-Format", Value: "binary"},
 		{Name: "Bundle-Version", Value: "2.0.0"},
-		{Name: "App-Name", Value: "Warp"},
-		// FIXME: Is this Action OK?
-		{Name: "Action", Value: "WarpMetaData"},
-		// Tags common with L2
-		{Name: "Sequencer", Value: "RedStone"},
+		// Global
+		{Name: "Source", Value: "Warp"},
```
Comment (szynwelski) [inline src/relay/arweave_meta_bundler.go:118]: {Name: "Source", Value: "Arweave"}?

File: src/relay/arweave_meta_bundler.go [hunk 5]
```diff
@@ -120,13 +111,21 @@ func (self *ArweaveMetaBundler) createMetaDataItem(payload *Payload, arweaveBloc
 	}
```
Comment (szynwelski) [inline src/relay/arweave_meta_bundler.go:111]: each L1 interaction should be wrapped in a separate meta data item

File: src/relay/msg_data_item_parser.go [hunk 6]
```diff
@@ -187,38 +188,33 @@ func (self *MsgDataItemParser) parseMessage(msg cosmostypes.Msg, payload *Payloa
 		State:          model.BundleStatePending,
 	}
 
-	contractId, ok := dataItem.DataItem.GetTag("Contract")
-	if !ok {
-		err = errors.New("failed to get contract id")
-		return
+	tags := []bundlr.Tag{
```
Comment (szynwelski) [inline src/relay/msg_data_item_parser.go:191]: add a common method that will set tags for meta data items of both types

File: src/relay/arweave_meta_bundler.go (hunk 4 above)
Comment (janekolszak) [reply src/relay/arweave_meta_bundler.go:118]: Done

File: src/relay/arweave_meta_bundler.go (hunk 5 above)
Comment (janekolszak) [reply src/relay/arweave_meta_bundler.go:111]: Why? The outcome in GQL is the same if we use a nested bundle

File: src/relay/arweave_meta_bundler.go (hunk 2 above)
Comment (janekolszak) [reply src/relay/arweave_meta_bundler.go:120]: Done

File: src/relay/arweave_meta_bundler.go (hunk 1 above)
Comment (janekolszak) [reply src/relay/arweave_meta_bundler.go:128]: Done

**Pull request merged**